### PR TITLE
PMM-7083 : change aws staging start to node based start

### PR DESCRIPTION
**Pull request by AlyHKafoury**

### CODE REVIEW

File: pmm/aws-staging-start.groovy [hunk 1]
```diff
@@ -315,35 +322,34 @@ pipeline {
...
+                        fi
+                        """
+                        node(env.VM_NAME){
+                            sh """
                                 set -o errexit
                                 set -o xtrace
-
                                 if [[ \$PMM_VERSION == pmm2 ]]; then
-                                    docker create \
+                                    sudo docker create \
```
Comment (EvgeniyPatlan) [inline pmm/aws-staging-start.groovy:347]: why do you need sudo for running docker?

File: pmm/aws-staging-start.groovy [hunk 2]
```diff
@@ -406,141 +411,139 @@ pipeline {
...
-                                sudo yum -y update
+                        set -o errexit
+                        set -o xtrace
+                        export PATH=\$PATH:/usr/sbin
+                        test -f /usr/lib64/libsasl2.so.2 || sudo ln -s /usr/lib64/libsasl2.so.3.0.0 /usr/lib64/libsasl2.so.2
+                        export CLIENT_IP=\$(curl ifconfig.me);
+                        sudo yum -y install https://repo.percona.com/yum/percona-release-latest.noarch.rpm
+                        if [[ \$CLIENT_VERSION = dev-latest ]]; then
+                            sudo percona-release disable all
+                            sudo percona-release enable original testing
```
Comment (EvgeniyPatlan) [inline pmm/aws-staging-start.groovy:424]:
```suggestion
                            sudo percona-release enable-only original testing
```

File: pmm/aws-staging-start.groovy [hunk 3]
```diff
@@ -406,141 +411,139 @@ pipeline {
...
+                            sudo yum clean all
+                            sudo yum makecache
+                            sudo yum -y install pmm2-client
+                            sudo yum -y update
+                        elif [[ \$CLIENT_VERSION = pmm2-latest ]]; then
+                            sudo yum clean all
+                            sudo yum -y install pmm2-client
+                            sudo yum -y update
+                            sudo percona-release disable all
+                            sudo percona-release enable original testing
```
Comment (EvgeniyPatlan) [inline pmm/aws-staging-start.groovy:434]:
```suggestion
                            sudo percona-release enable-only original testing
```

File: pmm/aws-staging-start.groovy [hunk 4]
```diff
@@ -406,141 +411,139 @@ pipeline {
...
+                        elif [[ \$CLIENT_VERSION = 2* ]]; then
+                            sudo yum clean all
+                            sudo yum -y install pmm2-client-\$CLIENT_VERSION-6.el6.x86_64
+                            sudo percona-release disable all
+                            sudo percona-release enable original testing
```
Comment (EvgeniyPatlan) [inline pmm/aws-staging-start.groovy:439]:
```suggestion
                            sudo percona-release enable-only original testing
```

File: pmm/aws-staging-start.groovy [hunk 5]
```diff
@@ -406,141 +411,139 @@ pipeline {
...
+                            sleep 15
+                        elif [[ \$CLIENT_VERSION = pmm1-dev-latest ]]; then
+                            sudo percona-release disable all
+                            sudo percona-release enable original testing
```
Comment (EvgeniyPatlan) [inline pmm/aws-staging-start.groovy:443]:
```suggestion
                            sudo percona-release enable-only original testing
```

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]
please provide link to the successfully finished job based on this pipeline

File: pmm/aws-staging-start.groovy [hunk 6]
```diff
@@ -201,6 +206,7 @@ pipeline {
                                 ],
                                 "EbsOptimized": false,
                                 "ImageId": "ami-15e9c770",
+                                "UserData": "c3VkbyB5dW0gaW5zdGFsbCAteSBqYXZhLTEuOC4wLW9wZW5qZGsKCnN1ZG8gL3Vzci9zYmluL2FsdGVybmF0aXZlcyAtLXNldCBqYXZhIC91c3IvbGliL2p2bS9qcmUtMS44LjAtb3Blbmpkay54ODZfNjQvYmluL2phdmEKCnN1ZG8gL3Vzci9zYmluL2FsdGVybmF0aXZlcyAtLXNldCBqYXZhYyAvdXNyL2xpYi9qdm0vanJlLTEuOC4wLW9wZW5qZGsueDg2XzY0L2Jpbi9qYXZhYwoKc3VkbyB5dW0gcmVtb3ZlIGphdmEtMS43Cg==",
```
Comment (atymchuk) [inline pmm/aws-staging-start.groovy:209]: Isn't this your personal credential?

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]
where is link to successful build with this pipeline

File: pmm/aws-staging-start.groovy (hunk 6 above)
Comment (AlyHKafoury) [reply pmm/aws-staging-start.groovy:209]: no this is commands to provision the instance you can echo them into `base64 -D` to decode them into the commands

**Comment by AlyHKafoury** [comment]
> where is link to successful build with this pipeline

@EvgeniyPatlan 
https://pmm.cd.percona.com/job/aws-staging-start/25296/console

**Review by EvgeniyPatlan — APPROVED** [review]
LGTM but please don't merge without @hors 's approval

File: pmm/aws-staging-start.groovy (hunk 6 above)
Comment (hors) [reply pmm/aws-staging-start.groovy:209]: why can not you do it via ?
https://pmm.cd.percona.com/configureClouds/
label awscli  -> Init script

Comment (AlyHKafoury) [reply pmm/aws-staging-start.groovy:209]: this is for nodes we create on the fly for QA engineers to use it is not jenkins slaves

**Review by atymchuk — APPROVED** [review]

File: pmm/aws-staging-start.groovy [hunk 7]
```diff
@@ -201,6 +206,7 @@ pipeline {
                                 ],
                                 "EbsOptimized": false,
                                 "ImageId": "ami-15e9c770",
```
Comment (hors) [inline pmm/aws-staging-start.groovy:208]: As I told earlier it is very old amazon linux image based on centos 6,  maybe if you update it to  amazon linux 2 which is based on centos 7 a lot of your issues will be fixed. Anyway you need to do it sooner or later.

Comment (atymchuk) [reply pmm/aws-staging-start.groovy:208]: Good point Slava, but we need this pipeline working. So we will have to address this upgrade in a separate PR, which will be dedicated to migration and maybe some refactoring thereof.

Comment (atymchuk) [reply pmm/aws-staging-start.groovy:208]: @hors Which image would be the best replacement for `ami-15e9c770`?

Comment (hors) [reply pmm/aws-staging-start.groovy:208]: @atymchuk try to find Amazon Linux 2 suitable for your jenkins (I mean in your region) using https://us-east-2.console.aws.amazon.com/ec2/v2/home?region=us-east-2#LaunchInstanceWizard e.g. this one ami-09558250a3419e7d0

File: pmm/aws-staging-start.groovy [hunk 8]
```diff
@@ -268,44 +274,54 @@ pipeline {
                 }
                 withCredentials([sshUserPrivateKey(credentialsId: 'aws-jenkins', keyFileVariable: 'KEY_PATH', passphraseVariable: '', usernameVariable: 'USER')]) {
                     sh """
-                        until ssh -i "${KEY_PATH}" -o ConnectTimeout=1 -o StrictHostKeyChecking=no ${USER}@\$(cat IP) date; do
+                        until ssh -i "${KEY_PATH}" -o ConnectTimeout=1 -o StrictHostKeyChecking=no ${USER}@\$(cat IP) 'java -version; sudo yum install -y java-1.8.0-openjdk; sudo /usr/sbin/alternatives --set java /usr/lib/jvm/jre-1.8.0-openjdk.x86_64/bin/java; java -version;' ; do
```
Comment (hors) [inline pmm/aws-staging-start.groovy:277]: Why do we need to install java  in these two places ?

https://github.com/Percona-Lab/jenkins-pipelines/pull/684/files#diff-cf39489e2a5e3b641bec0e09fa1ee7a4f9f4c076eb1b8a7aa517440fb2e22a9cR209
https://github.com/Percona-Lab/jenkins-pipelines/pull/684/files#diff-cf39489e2a5e3b641bec0e09fa1ee7a4f9f4c076eb1b8a7aa517440fb2e22a9cR277

And why do we need to do it at all?  We are connecting to this remote instance  via ssh and it is not jenkins worker. 

Also I do not see a good commit message in this PR. What was the root of issue? What do we try to fix? 

As for me for now it seems as a workaround but not a general fix but I can be wrong. And as I know it is pretty urgent so please update the description and commit and then re-check everything without any rush.

**Review by hors — APPROVED** [review]

**Review by puneet0191 — APPROVED** [review]

**Pull request merged**